amlb/uploads.py

```diff
@@ -0,0 +1,197 @@
+import json
+import logging
+import pathlib
+from collections import OrderedDict
+import textwrap
+from typing import Set, Optional, List
+
+import openml
+import pandas as pd
+from openml import OpenMLTask, OpenMLFlow
+from openml.runs.functions import format_prediction
+
+from .utils.core import Namespace
+
+log = logging.getLogger(__name__)
+
+
+def _load_task_data(task_folder: pathlib.Path, fold: int = 0) -> Namespace:
+    """ Loads the metadata of the given fold of a task as a namespace. """
+    with open(task_folder / str(fold) / 'metadata.json', 'r') as fh:
+        metadata = json.load(fh)
+    metadata = Namespace.from_dict(metadata)
+    return metadata
+
+
+def _load_fold(task_folder: pathlib.Path, fold: int, task: OpenMLTask) -> pd.DataFrame:
+    """ Load the predictions and add openml repeat/fold/index information. """
+    prediction_file = task_folder / f"{fold}/predictions.csv"
+    predictions = pd.read_csv(prediction_file, sep=',', header=0)
+
+    train_indices, test_indices = task.get_train_test_split_indices(fold, repeat=0, sample=0)
+    predictions["index"] = test_indices
+    predictions["fold"] = fold
+    predictions["repeat"] = 0
+    return predictions
+
+
+def _load_predictions(task_folder: pathlib.Path) -> pd.DataFrame:
+    """ Loads predictions of all folds for a task with index information required for upload. """
+    metadata = _load_task_data(task_folder)
+    task = openml.tasks.get_task(metadata.openml_task_id)
+    results = [_load_fold(task_folder, fold, task) for fold in range(10)]
+    return pd.concat(results)
+
+
+def _list_completed_folds(task_folder: pathlib.Path) -> Set[str]:
+    completed_folds = set()
+    for fold_dir in [d for d in task_folder.iterdir() if d.is_dir()]:
+        if (task_folder / fold_dir / "predictions.csv").is_file():
+            completed_folds.add(fold_dir.name)
+    return completed_folds
+
+
+def _get_flow(metadata: Namespace, sync_with_server: bool = True) -> openml.flows.OpenMLFlow:
+    """ Creates or retrieves an OpenML flow for the given run metadata. """
+    if metadata.git_info.repo != "NA":
+        amlb_version = metadata.git_info.commit[:8] if metadata.git_info.tags == [] else metadata.git_info.tags[0]
```

Comment:
This is a very specific check for the tag being an empty list. Maybe it is sufficient to just check `if metadata.git_info.tags else`?